### PR TITLE
add: rand window support respect|ignore nulls

### PR DESCRIPTION
**Pull request by TCeason**

### CODE REVIEW

**Comment by TCeason** [comment]
yarn run dev is ok on local.

File: docs/en/sql-reference/20-sql-functions/08-window-functions/first-value.md [hunk 1]
```diff
@@ -16,11 +16,17 @@ See also:
...
 ```sql
-FIRST_VALUE(expression) OVER ([PARTITION BY partition_expression] ORDER BY order_expression [window_frame])
+FIRST_VALUE(expression) [ { IGNORE | RESPECT } NULLS ] OVER ([PARTITION BY partition_expression] ORDER BY order_expression [window_frame])
 ```
 
 For the syntax of window frame, see [Window Frame Syntax](index.md#window-frame-syntax).
 
+## Usage notes
+
+When this option is used with FIRST_VALUE, the function returns the first value in the frame that is not NULL (or NULL if all values are NULL).
```
Comment (soyeric128) [inline docs/en/sql-reference/20-sql-functions/08-window-functions/first-value.md:26]:
```suggestion
When `{ IGNORE | RESPECT } NULLS` is used with FIRST_VALUE, the function returns the first value in the frame that is not NULL (or NULL if all values are NULL).
```

File: docs/en/sql-reference/20-sql-functions/08-window-functions/first-value.md [hunk 2]
```diff
@@ -16,11 +16,17 @@ See also:
...
+
+If { IGNORE | RESPECT } NULLS is not specified, the default is RESPECT NULLS.
```
Comment (soyeric128) [inline docs/en/sql-reference/20-sql-functions/08-window-functions/first-value.md:28]: Moved to the line above, so we can now get rid of this line.

File: docs/en/sql-reference/20-sql-functions/08-window-functions/first-value.md [hunk 3]
```diff
@@ -52,4 +58,41 @@ employee_id | first_name | last_name | salary  | highest_salary_first_name
 3           | David      | Johnson   | 5500.00 | Mary
 1           | John       | Doe       | 5000.00 | Mary
 5           | Michael    | Brown     | 4500.00 | Mary
-```
\ No newline at end of file
+
+```
+
+### Use IGNORE NULLS to find non-NULL values
```
Comment (soyeric128) [inline docs/en/sql-reference/20-sql-functions/08-window-functions/first-value.md:64]:
```suggestion
### Returning NON-NULLs with IGNORE NULLS 

```

File: docs/en/sql-reference/20-sql-functions/08-window-functions/last-value.md [hunk 4]
```diff
@@ -16,11 +16,17 @@ See also:
...
 ```sql
-LAST_VALUE(expression) OVER ([PARTITION BY partition_expression] ORDER BY order_expression [window_frame])
+LAST_VALUE(expression) [ { IGNORE | RESPECT } NULLS ] OVER ([PARTITION BY partition_expression] ORDER BY order_expression [window_frame])
 ```
 
 For the syntax of window frame, see [Window Frame Syntax](index.md#window-frame-syntax).
 
+## Usage notes
+
+When this option is used with LAST_VALUE, The function returns the last value in the frame that is not NULL (or NULL if all values are NULL).
```
Comment (soyeric128) [inline docs/en/sql-reference/20-sql-functions/08-window-functions/last-value.md:26]:
```suggestion
When `{ IGNORE | RESPECT } NULLS` is used with LAST_VALUE, the function returns the last value in the frame that is not NULL (or NULL if all values are NULL). If not specified, the default is RESPECT NULLS.

```

File: docs/en/sql-reference/20-sql-functions/08-window-functions/last-value.md [hunk 5]
```diff
@@ -16,11 +16,17 @@ See also:
...
+
+If { IGNORE | RESPECT } NULLS is not specified, the default is RESPECT NULLS.
```
Comment (soyeric128) [inline docs/en/sql-reference/20-sql-functions/08-window-functions/last-value.md:28]: Delete

File: docs/en/sql-reference/20-sql-functions/08-window-functions/last-value.md [hunk 6]
```diff
@@ -51,4 +57,40 @@ employee_id | first_name | last_name | salary  | lowest_salary_first_name
 3           | David      | Johnson   | 5500.00 | Michael
 1           | John       | Doe       | 5000.00 | Michael
 5           | Michael    | Brown     | 4500.00 | Michael
-```
\ No newline at end of file
+```
+
+### Use IGNORE NULLS to find non-NULL values
```
Comment (soyeric128) [inline docs/en/sql-reference/20-sql-functions/08-window-functions/last-value.md]:
```suggestion
### Returning NON-NULLs with IGNORE NULLS

```

File: docs/en/sql-reference/20-sql-functions/08-window-functions/nth-value.md [hunk 7]
```diff
@@ -4,9 +4,9 @@ title: NTH_VALUE
 
 import FunctionDescription from '@site/src/components/FunctionDescription';
 
-<FunctionDescription description="Introduced: v1.1.50"/>
+<FunctionDescription description="Introduced: v1.2.568"/>
 
-Returns the Nth value from an ordered group of values.
+Returns expr evaluated at the nth row (among rows with a non-null value of expr if `IGNORE NULLS` is set) of the window frame (counting from 1); NULL if no such row.
```
Comment (soyeric128) [inline docs/en/sql-reference/20-sql-functions/08-window-functions/nth-value.md:9]:
```suggestion
Returns expr evaluated at the nth row (starting from 1), or NULL if the row does not exist at all.  If `IGNORE NULLS` is specified, the function ignores the rows with NULLs.
```

**Review by soyeric128 — APPROVED** [review]

**Pull request merged**